jarbas/core/management/commands/reimbursements.py

```diff
@@ -31,53 +34,31 @@ def handle(self, *args, **options):
     @property
     def reimbursements(self):
         """Returns a Generator with a dict object for each row."""
-        with lzma.open(self.path, mode='rt') as file_handler:
-            for row in csv.DictReader(file_handler):
-                yield self.serialize(row)
-
-    def serialize(self, reimbursement):
-        """Read the dict generated by DictReader and fix content types"""
-
-        missing = ('probability', 'suspicions')
-        for key in missing:
-            reimbursement[key] = None
-
+        force_types = {
+            'cnpj_cpf': TextField,
+            'document_number': TextField,
+            'leg_of_the_trip': TextField,
+            'congressperson_id': IntegerField,
+            'congressperson_document': IntegerField,
+            'reimbursement_value_total': FloatField,
+            'issue_date': DateField,
+            'term': IntegerField,
+            'term_id': IntegerField
+        }
+        with lzma.open(self.path) as file_handler:
+            for row in import_from_csv(file_handler, force_types=force_types):
+                as_dict = dict(row._asdict())  # _asdict returns OrderedDict
+                yield self.rename(as_dict)
```

Comment:
You can use `yield from` and centralize all transformation in one method (instead of having a separate method to rename only - this way you can do other transformations as well (if needed) and use this API as a standard to convert data rows). I'd refactor this to:

```python
        with lzma.open(self.path) as file_handler:
            yield from map(self.transform_row,
                           import_from_csv(file_handler,
                                           force_types=force_types))

    @staticmethod
    def transform_row(row):
        """Read the dict generated by rows and fix some keys"""
        rename = {
                'subquota_number': 'subquota_id',
                'reimbursement_value_total': 'total_reimbursement_value',
        )
        return {rename.get(key, key): getattr(row, key)
                for key, value in rename.items()}
    # and remove `rename` method
```

Reply:
As we're using Celery to process it now, I just converted the `row` to a dictionary and passed it to be processed asynchronously. However I adopted the `transform_row` semantics in `tasks.py` ; ) Tks!